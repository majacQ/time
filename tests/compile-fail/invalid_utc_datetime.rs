use time::macros::utc_datetime;

fn main() {
    let _ = utc_datetime!(2021-000 0:00);
    let _ = utc_datetime!(2021-001 24:00);
    let _ = utc_datetime!(2021-001 0:00 0);
    let _ = utc_datetime!(2021-001 0:00 UTC);
    let _ = utc_datetime!(2021-001 0:00 UTC x);
}
