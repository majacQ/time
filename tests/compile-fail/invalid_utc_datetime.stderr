error: invalid component: ordinal was 0
 --> $WORKSPACE/tests/compile-fail/invalid_utc_datetime.rs
  |
  |     let _ = utc_datetime!(2021-000 0:00);
  |                                ^^^

error: invalid component: hour was 24
 --> $WORKSPACE/tests/compile-fail/invalid_utc_datetime.rs
  |
  |     let _ = utc_datetime!(2021-001 24:00);
  |                                    ^^

error: unexpected token: 0
 --> $WORKSPACE/tests/compile-fail/invalid_utc_datetime.rs
  |
  |     let _ = utc_datetime!(2021-001 0:00 0);
  |                                         ^

error: unexpected token: UTC
 --> $WORKSPACE/tests/compile-fail/invalid_utc_datetime.rs
  |
  |     let _ = utc_datetime!(2021-001 0:00 UTC);
  |                                         ^^^

error: unexpected token: UTC
 --> $WORKSPACE/tests/compile-fail/invalid_utc_datetime.rs
  |
  |     let _ = utc_datetime!(2021-001 0:00 UTC x);
  |                                         ^^^
